File: test_db.py
import sqlite3
import json

conn = sqlite3.connect('apps/worker/data/conversion_jobs.db')
c = conn.cursor()
c.execute("SELECT id, crawl_artifacts, extraction_report FROM conversion_jobs LIMIT 3")
for row in c.fetchall():
    job_id, ca_str, er_str = row
    ca = json.loads(ca_str) if ca_str else {}
    er = json.loads(er_str) if er_str else {}
    print(f"Job: {job_id}")
    print(f"crawl_artifacts.assets type: {type(ca.get('assets'))}")
    if ca.get('assets') is not None:
        print(f"crawl_artifacts.assets length: {len(ca.get('assets'))}")
    print(f"crawl_artifacts.totals: {ca.get('totals')}")
    print(f"extraction_report.assets type: {type(er.get('assets'))}")

File: test_db.js
const axios = require('axios');
const sqlite3 = require('sqlite3');

const db = new sqlite3.Database('apps/worker/data/conversion_jobs.db', (err) => {
  if (err) console.error(err.message);
});

db.all("SELECT id, crawl_artifacts, extraction_report FROM conversion_jobs LIMIT 3", [], (err, rows) => {
  if (err) throw err;
  rows.forEach(row => {
    const ca = JSON.parse(row.crawl_artifacts || '{}');
    const er = JSON.parse(row.extraction_report || '{}');
    console.log(`Job: ${row.id}`);
    console.log(`crawl_artifacts.assets type:`, Array.isArray(ca.assets) ? 'array' : typeof ca.assets);
    if (ca.assets) console.log(`crawl_artifacts.assets length:`, ca.assets.length);
    console.log(`crawl_artifacts.totals:`, ca.totals);
    console.log(`extraction_report.assets type:`, Array.isArray(er.assets) ? 'array' : typeof er.assets);
  });
});
